Engine/Results/BacktestingResultHandler.cs

```diff
@@ -865,18 +865,36 @@ public void ProcessSynchronousEvents(bool forceProcess = false)
                 }
             }
 
-            //Send out the debug messages:
-            _algorithm.DebugMessages.ForEach(x => DebugMessage(x));
-            _algorithm.DebugMessages.Clear();
+            //Send out the debug messages:            
+            while (_algorithm.DebugMessages.Count > 0)
```

Comment:
This could potentially hang forever. We should make a timeout for the loops or restructure it to copy to a list first before sending.